### DISCUSSION TITLE
How can one use Prisma with NestJS Swagger annotations?

### DISCUSSION THREAD

**Discussion by khalibloo** [prisma/prisma#6197] [Q&A]
NestJS's swagger integration relies on DTO classes to generate types for the API docs. However, prisma generates typescript types rather than classes. And these cannot be annotated.  Is there a way to make prisma generate DTO classes instead of types?

Here is an example of what NestJS expects.
```ts
export class CreateUserDto {
  email: string;
  password: string;
  roles: RoleEnum[] = [];
  isEnabled?: boolean = true;
}
```

I've tried making DTO classes that implement the prisma types, but it's easy to introduce errors and inconsistencies in the API docs doing this manually.

**Comment by khalibloo** [comment]
After some more digging, it appears others have asked similar questions about classes. I suppose prisma just doesn't suit this kind of use case.

**Reply by ryands17** [reply]
@khalibloo 👋 

Currently it's not as easy, but I would suggest watching this request: https://github.com/prisma/prisma/issues/5273 and looking at the workarounds mentioned.

Also there's an explicit request for the same https://github.com/prisma/prisma/issues/5273 that you can follow :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
